Add some missing move animations to the move animation tests

FosterProgramming opened the pull request:
The move animations tests used to only play the first part of two turn moves like "Fly", "Dig" or "Solar Beam".
This PR fixes that so that.

AlexOn1ine commented:
you can set the pr to draft. that way you don't need to change the title. 
<img width="349" height="204" alt="image" src="https://github.com/user-attachments/assets/b6cc9fe9-deee-47e1-bcc9-f72f27673d34" />

FosterProgramming commented:
I tried to find all the animations with "branches" and I made sure the tests will play every variation of the animation.
Most of the test changes are straightforward but if you find it overwhelming to look at all of them at once, I would recommend doing the review commit by commit.
Three notables changes outside of the move animations test:
- i changed the random functions used in the moves magnitude and present to use a random tag
- I modified the code that creates test pokemon to allow the definition of shiny pokemon inside tests

FosterProgramming commented:
The shell side arm commit was tested independantly of other moves and does not work in the test suite because of a RNG_SHELL_SIDE_ARM bug describe in issue #7547

AlexOn1ine commented on test/battle/move_animations/all_anims.c:
```diff
@@ -104,7 +198,38 @@ static bool32 UserHasToGoFirst(u32 move) // Player needs to go first
+{
+    u32 variationsNumber;
+
+    if (gMovesInfo[move].effect == EFFECT_WEATHER_BALL
+        || gMovesInfo[move].effect == EFFECT_TERRAIN_PULSE
+        || move == MOVE_TECHNO_BLAST)
+        variationsNumber = 5;
+    else if (gMovesInfo[move].effect == EFFECT_FRUSTRATION
+        || gMovesInfo[move].effect == EFFECT_RETURN
+        || gMovesInfo[move].effect == EFFECT_IVY_CUDGEL
+        || move == MOVE_WATER_SPOUT //we don't use the effect because other moves with the water spout effect don't have animation variations
+        )
+        variationsNumber = 4;
+    else if (gMovesInfo[move].effect == EFFECT_SPIT_UP
+        || gMovesInfo[move].effect == EFFECT_SWALLOW
+        || move == MOVE_SEISMIC_TOSS
+        || gMovesInfo[move].effect == EFFECT_DRAGON_DARTS)
+        variationsNumber = 3;
+    else if (gMovesInfo[move].effect == EFFECT_CURSE
+        || gMovesInfo[move].effect == EFFECT_PRESENT
+        || (isDouble && gMovesInfo[move].effect == EFFECT_TERA_STARSTORM)
+        || move == MOVE_SYRUP_BOMB
+        || gMovesInfo[move].effect == EFFECT_SHELL_SIDE_ARM
+        || gMovesInfo[move].effect == EFFECT_FICKLE_BEAM
+        || gMovesInfo[move].effect == EFFECT_MAGNITUDE)
```
```suggestion
    if (gMovesInfo[move].effect == EFFECT_WEATHER_BALL
     || gMovesInfo[move].effect == EFFECT_TERRAIN_PULSE
     || move == MOVE_TECHNO_BLAST)
        variationsNumber = 5;
    else if (gMovesInfo[move].effect == EFFECT_FRUSTRATION
          || gMovesInfo[move].effect == EFFECT_RETURN
          || gMovesInfo[move].effect == EFFECT_IVY_CUDGEL
          || move == MOVE_WATER_SPOUT) //we don't use the effect because other moves with the water spout effect don't have animation variations
        variationsNumber = 4;
    else if (gMovesInfo[move].effect == EFFECT_SPIT_UP
          || gMovesInfo[move].effect == EFFECT_SWALLOW
          || move == MOVE_SEISMIC_TOSS
          || gMovesInfo[move].effect == EFFECT_DRAGON_DARTS)
        variationsNumber = 3;
    else if (gMovesInfo[move].effect == EFFECT_CURSE
          || gMovesInfo[move].effect == EFFECT_PRESENT
          || (isDouble && gMovesInfo[move].effect == EFFECT_TERA_STARSTORM)
          || move == MOVE_SYRUP_BOMB
          || gMovesInfo[move].effect == EFFECT_SHELL_SIDE_ARM
          || gMovesInfo[move].effect == EFFECT_FICKLE_BEAM
          || gMovesInfo[move].effect == EFFECT_MAGNITUDE)
```

AlexOn1ine commented:
@FosterProgramming could you solve conflicts please? It can be merged after.
